pydbgen/pydbgen.py

```diff
@@ -1,41 +1,65 @@
+import os
+import random
+from random import randint, choice
+import pandas as pd
+
 class pydb():
-    
-    def __init__(self,seed=None):
+
+    def __init__(self, seed=None):
         """
         Initiates the class and creates a Faker() object for later data generation by other methods
         seed: User can set a seed parameter to generate deterministic, non-random output
         """
         from faker import Faker
-        import pandas as pd
-        from random import randint,choice
-        
-        self.fake=Faker()
-        self.seed=seed
-        self.randnum=randint(1,9)
 
+        self.fake = Faker()
+        self.seed = seed
+        self.randnum = randint(1, 9)
+
+        self.city_list = self._initialize_city_list()
+        self.domain_list = self._initialize_email_domain_list()
+
+    def _initialize_city_list(self):
+        from six import moves
+        import ssl
+
+        path = "US_Cities.txt"
+        if not os.path.isfile(path):
+            context = ssl._create_unverified_context()
+            moves.urllib.request.urlretrieve("https://raw.githubusercontent.com/tflearn/tflearn.github.io/master/resources/US_Cities.txt", path)
+
+        city_list = []
+        with open(path) as fh:
+            city_list = [str(line).strip() for line in fh.readlines()]
+
+        return city_list
 
-    def simple_ph_num(self,seed=None):
+    def _initialize_email_domain_list(self):
+        dir_path = os.path.dirname(os.path.realpath(__file__))
+        path = dir_path + os.sep + "Domains.txt"
+
+        domain_list = []
+        with open(path) as fh:
+            domain_list = [str(line).strip() for line in fh.readlines()]
+
+        return domain_list
+
+    def simple_ph_num(self, seed=None):
         """
         Generates 10 digit US phone number in xxx-xxx-xxxx format
         seed: Currently not used. Uses seed from the pydb class if chosen by user
         """
-        import random
-        from random import randint,choice
         random.seed(self.seed)
 
-        result = str(randint(1,9))

```

Comment:
String appends are slow since strings are immutable in Python. For small datasets, this wouldn't matter, but it will immediately slow down attempts to generate larger ones.